source/common/http/filter/fault_filter.cc

```diff
@@ -142,6 +150,21 @@ void FaultFilter::abortWithHTTPStatus() {
   callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::FaultInjected);
 }
 
+bool FaultFilter::matchesTargetCluster(HeaderMap& headers) {
+  bool matches = true;
+
+  if (!config_->targetCluster().empty()) {
+    // TODO PERF Eliminate repeated route matching
```

Comment:
I would take out the comment and just open a GH issue to remind us to fix this. The fix is small now that we have the stable route table interface. The selected route can just be stored inside of ActiveStream and returned if it was already looked up.